src/190B20.c

```diff
@@ -1094,9 +1120,75 @@ void func_8026324C(s32 arg0, s32 arg1) {
     func_80263064(arg0, arg1, 1);
 }
 
-INCLUDE_ASM(s32, "190B20", func_80263268);
+void func_80263268(void) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    PlayerData* playerData = &gPlayerData;
+    Actor* partner = battleStatus->partnerActor;
+
+    battleStatus->changePartnerAllowed = 0;
+    if (partner != NULL) {
+        s32 partnersEnabled;
+        s32 i;
+
+        battleStatus->changePartnerAllowed = 1;
+        partnersEnabled = 0;
+
+        for (i = 0; i < ARRAY_COUNT(playerData->partners); i++) {
+            if (playerData->partners[i].enabled) {
+                partnersEnabled++;
+            }
+        }
 
-INCLUDE_ASM(s32, "190B20", func_80263300);
+        if (partnersEnabled >= 2) {
+            if (partner->koStatus == STATUS_DAZE) {
+                battleStatus->changePartnerAllowed = 0;
+            } else if (partner->debuff == 7) {
```

Comment:
Use enum

Reply:
Which enum? I don't think one exists for this

Reply:
nm it's STATUS_